packages/cashier/src/components/percentage-selector/percentage-selector.tsx

```diff
@@ -41,41 +55,67 @@ const PercentageSelector = ({
         getCalculatedAmount((amount * (new_percentage / 100)).toFixed(getDecimalPlaces(currency)));
 
         for (let i = 1; i <= 4; i++) {
-            if (i < e.target.id || (i === +e.target.id && !is_percentage_selected)) {
-                document.getElementById(i).style.backgroundColor = 'var(--status-success)';
-            } else {
-                document.getElementById(i).style.backgroundColor = 'var(--general-section-1)';
+            const percentage_selector_block = document.getElementById(String(i));
+            if (percentage_selector_block) {
+                if (
+                    i < (e as TCalculateAmountInputEvent).target.id ||
+                    (i === +(e as TCalculateAmountInputEvent).target.id && !is_percentage_selected)
+                ) {
+                    percentage_selector_block.style.backgroundColor = 'var(--status-success)';
+                } else {
+                    percentage_selector_block.style.backgroundColor = 'var(--general-section-1)';
+                }
             }
         }
     };
     const format_amount = formatMoney(currency, amount, true);
     const currency__display_code = getCurrencyDisplayCode(currency);
     return (
         <React.Fragment>
-            <div className='percentage-selector'>
+            <div className='percentage-selector' data-testid='percentage-selector-id'>
```

Comment:
@vinu-deriv Could you update the data-testids base on the naming convention we have: https://wikijs.deriv.cloud/Frontend/General-information/test_guide#:~:text=Make%20sure%20you%20apply%20correct%20names%20while%20using%20data%2Dtestid%20attribute.%20It%20is%20required%20to%20use%20snake_case%20and%20dt%20prefix%3A